app-frontend/src/app/components/histogram/quickEditHistogram/index.js

```diff
@@ -0,0 +1,289 @@
+import tpl from './index.html';
+import { get, debounce, min, max, keys, first, last, entries, sortBy } from 'lodash';
+
+import {
+    breakpointsFromRenderDefinition,
+    renderDefinitionFromState,
+    colorStopsToRange
+} from '_redux/histogram-utils';
+
+class QuickEditHistogramController {
+    constructor($rootScope, uuid4, $q, $element, $window, $timeout, graphService) {
+        'ngInject';
+        $rootScope.autoInject(this, arguments);
+    }
+
+    $onInit() {
+        this.graphId = this.uuid4.generate();
+        if (!this.currentUpdate && this.plot && this.renderDefinition && this.ranges) {
+            this.updateGraph(this.plot, this.renderDefinition, this.ranges);
+        }
+        this.debouncedBreakpointChange = debounce(this.onBreakpointChange.bind(this), 500);
+        this.initGraph();
+        this.onWindowResizeDebounced = debounce(this.onWindowResize.bind(this), 500);
+        this.$window.addEventListener('resize', this.onWindowResizeDebounced.bind(this));
+        this.$timeout(this.onWindowResize.bind(this), 500);
+    }
+
+    $onDestroy() {
+        if (this.onWindowResize) {
+            this.$window.removeEventListener('resize', this.onWindowResizeDebounced);
+        }
+        this.graphService.deregister(this.graphId);
+    }
+
+    onWindowResize() {
+        this.getGraph().then(graph => {
+            graph.render();
+        });
+    }
+
+    $onChanges(changes) {
+        const plot = get(changes, 'plot.currentValue');
+        const metadata = get(changes, 'metadata.currentValue');
+        const renderDefinition = get(metadata, 'renderDefinition');
+        if (renderDefinition) {
+            this.renderDefinition = renderDefinition;
+        }
+        const histogramOptions = get(metadata, 'histogramOptions');
+        if (histogramOptions) {
+            this.histogramOptions = histogramOptions;
+        }
+        this.options = Object.assign(
+            {},
+            this.options,
+            this.histogramOptions
+                ? {
+                      dataRange: this.histogramOptions.range,
+                      baseScheme: this.histogramOptions.baseScheme
+                  }
+                : {}
+        );
+        if (this.renderDefinition) {
+            this.breakpoints = breakpointsFromRenderDefinition(
+                this.renderDefinition,
+                this.uuid4.generate
+            );
+        }
+        const ranges = get(changes, 'ranges.currentValue');
+        if (plot || metadata || ranges) {
```

Comment:
`plot || metadata || ranges` are the changed items, renderDefinition is derivative from metadata. We don't want to call `updateGraph(...)` unless all 3 are defined, but we want to check whenever 1 of the 3 are changed. I suppose it would make more sense to just move the check for the graph into `updateGraph(...)`.